akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventsourcedRunning.scala

```diff
@@ -161,12 +164,12 @@ import scala.collection.immutable
       if (tags.isEmpty) event else Tagged(event, tags)
     }
 
-    withMdc("run-cmnds") {
+    withMdc(setup, MDC.RunningCmds) {
       Behaviors.immutable[EventsourcedBehavior.InternalProtocol] {
         case (_, IncomingCommand(c: C @unchecked)) ⇒ onCommand(state, c)
-        case (_, SnapshotterResponse(r))           ⇒ Behaviors.unhandled
-        case (_, JournalResponse(r))               ⇒ Behaviors.unhandled
-      }
+        case (_, SnapshotterResponse(_))           ⇒ Behaviors.unhandled
+        case (_, JournalResponse(_))               ⇒ Behaviors.unhandled
+      }.onSignal(returnPermitOnStop)
```

Comment:
In Running this shouldn't be needed because the permit is always returned before that phase.

Reply:
right, thanks